Gemfile

```diff
@@ -85,6 +86,11 @@ group :heroku do
   gem 'rails_12factor'
 end
 
+gem 'rails', github: 'rails', branch: '5-2-stable'
+gem 'cancancan', github: 'lizzyaustad/cancancan', branch: 'rails_5_2_upgrade_fixes'
```

Comment:
https://github.com/CanCanCommunity/cancancan/blob/develop/CHANGELOG.md#220-apr-13th-2018 - can pin to < 3.0.0